shell/platform/android/io/flutter/embedding/engine/FlutterEngine.java

```diff
@@ -378,6 +389,7 @@ public void destroy() {
     flutterJNI.removeEngineLifecycleListener(engineLifecycleListener);
     flutterJNI.setDynamicFeatureManager(null);
     flutterJNI.detachFromNativeAndReleaseResources();
+    dynamicFeatureChannel.destroy();
```

Comment:
maybe just do if feature manager is not null, set channel to null in the if below? So we don't have to create a new channel destroy pattern.